src/structs/bit_writer.rs

```diff
@@ -44,8 +43,29 @@ impl BitWriter {
         self.current_bit = tmp_current_bit;
     }
 
+    #[inline(never)]
```

Comment:
Please add comments why it should be a separate method, why it's good for perf to have a "cold" method which is almost a duplicate of another method.